### PR TITLE
[BugFix] Fix dead lock of primary key table

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

We may get dead lock for PrimaryKey table in the following scenario：
1. The first thread is doing primary clone task and call function `TabletUpdates::load_snapshot` which hold `_lock` of `TabletUpdates`. In the end of this function, we will release the primary index in index cache and call function `PrimaryIndex::unload` which acquire `_lock` in PrimaryIndex.
2. The second thread is doing data import and call function `UpdateManager::on_rowset_finished` in the commit phase. If we don't get primary index in the index cache, we will load primary index and call function `PrimaryIndex::load` which hold the `_lock` of PrimaryIndex. After that, we need to get applied rowset to build index which acuqire `_lock` in TabletUpdates.

Actually, when we finished load snapshot, the only thing we need to do is release the primary index and we don't need to hold the `_lock` of TabletUpdates anymore, so we can unlock in advance to avoid dead lock.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

**Review by trueeyu — DISMISSED** [review]

**Review by decster — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -3256,16 +3256,17 @@ Status TabletUpdates::load_snapshot(const SnapshotMeta& snapshot_meta, bool rest
         l2.unlock();                     // _rowsets_lock
         _try_commit_pendings_unlocked(); // may acquire |_rowset_stats_lock| and |_rowsets_lock|
 
+        _update_total_stats(_edit_version_infos[_apply_version_idx]->rowsets, nullptr, nullptr);
+        _apply_version_changed.notify_all();
+        l1.unlock();
```
Comment (imay) [inline be/src/storage/tablet_updates.cpp:3261]: I will be better to add some comments in the code base to explain why the lock is unlock here.

Comment (sevev) [reply be/src/storage/tablet_updates.cpp:3267]: > I will be better to add some comments in the code base to explain why the lock is unlock here.

done, add some comments in https://github.com/StarRocks/starrocks/pull/18488/commits/6c71ec140154bfcb073ec4f08489c78dcf05108b

**Comment by sevev** [comment]
run starrocks_clang-format

**Comment by sevev** [comment]
run starrocks_be_unittest

**Review by decster — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**